### DISCUSSION TITLE
Forwardable - def_delegator nasıl çalışır?

### DISCUSSION THREAD

**Discussion by ipekozbay** [19/classroom#51] [Ruby] [answered]
Forwardable modülünü çağırdığımızı aşağıdaki kodda def delegator ' ın ne yaptığını anlayamadım. Ayrıca bu modül bu kodda ne işimize yaradı? 
<img width="607" alt="Ekran Resmi 2022-06-29 19 44 49" src="https://user-images.githubusercontent.com/72815766/176491985-deaaa4d9-13a1-4d8d-b707-1cabb98c6aaa.png">

**Comment by roktas** [comment, marked as answer]
> Forwardable modülünü çağırdığımızı aşağıdaki kodda def delegator ' ın ne yaptığını anlayamadım.

Ne yaptığını buradan öğrenebilirsin: https://ruby-doc.org/stdlib-2.7.1/libdoc/forwardable/rdoc/Forwardable.html#method-i-def_delegator

> Ayrıca bu modül bu kodda ne işimize yaradı?

`Rectangular` sınıfında `distance` adında bir metod var mı?  Yoksa nasıl geldi?  (Geldiyse işimize yaradı)

**Reply by ipekozbay** [reply]
Teşekkür ederim hocam anladım.

**Comment by roktas** [comment]
Ah bir de bu çalışmaları son güne bırakmasanız...
